staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -110,37 +110,13 @@ func newPod() runtime.Object {
 }
 
 func TestCreate(t *testing.T) {
-	ctx, store, etcdClient := testSetup(t)
-
-	key := "/testkey"
-	out := &example.Pod{}
-	obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", SelfLink: "testlink"}}
-
-	// verify that kv pair is empty before set
-	getResp, err := etcdClient.KV.Get(ctx, key)
-	if err != nil {
-		t.Fatalf("etcdClient.KV.Get failed: %v", err)
-	}
-	if len(getResp.Kvs) != 0 {
-		t.Fatalf("expecting empty result on key: %s", key)
-	}
-
-	err = store.Create(ctx, key, obj, out, 0)
-	if err != nil {
-		t.Fatalf("Set failed: %v", err)
-	}
-	// basic tests of the output
-	if obj.ObjectMeta.Name != out.ObjectMeta.Name {
-		t.Errorf("pod name want=%s, get=%s", obj.ObjectMeta.Name, out.ObjectMeta.Name)
-	}
-	if out.ResourceVersion == "" {
-		t.Errorf("output should have non-empty resource version")
-	}
-	if out.SelfLink != "" {
-		t.Errorf("output should have empty selfLink")
-	}
+	_, store, etcdClient := testSetup(t)
```

Comment:
Copying from PR description:
```
This is a first step (just for Create method) towards unifying tests across etcd3 and watchache.

Given they both implement the same interface, we can reasonably expect they implement the same semantic.
Unifying the tests across both implementations would allow us to catch and fix any potential differences

This will also be useful to clearly define the expected semantic of any storage interface.
```

Before I do more of that, I would like to get feedback about this approach.

@liggitt @deads2k @smarterclayton @jpbetz - for thoughts